src/App.jsx

```diff
@@ -1,15 +1,64 @@
 import { hot } from 'react-hot-loader';
-import React from 'react';
+import React, { useState } from 'react';
+import { Authorize } from 'react-auth0-components';
 import { ThemeProvider } from '@material-ui/styles';
+import AuthContext from './utils/AuthContext';
+import { USER_SESSION } from './utils/constants';
 import theme from './theme';
 import Main from './Main';
 
-const App = () => (
-  <div>
-    <ThemeProvider theme={theme}>
-      <Main />
-    </ThemeProvider>
-  </div>
-);
+const App = () => {
+  const [authorize, setAuthorize] = useState(false);
+  const [authContext, setAuthContext] = useState({
+    authorize: () => setAuthorize(true),
+    unauthorize: () => {
+      setAuthorize(false);
+      setAuthContext({
+        ...authContext,
+        user: null,
+      });
+    },
+    user: null,
+  });
+
+  return (
+    <AuthContext.Provider value={authContext}>
+      <ThemeProvider theme={theme}>
+        <Authorize
+          authorize={authorize}
+          popup
```

Comment:
Can we do this in a new tab instead of a new window? It's a little less jarring that way.

Reply:
If we do it in a new tab then we won't be able to use `react-auth-components` unfortunately since the refresh token rely on the `Authorize` component so that it can be updated.

Reply:
Okay, not a blocker.